x/tss/keeper/keeperKeygen.go

```diff
@@ -174,8 +183,13 @@ func (k Keeper) AssignNextKey(ctx sdk.Context, chain nexus.Chain, keyRole export
 	// The key entry needs to store the keyID instead of the public key, because the keyID is needed whenever
 	// the keeper calls the secure private key store (e.g. for signing) and we would lose the keyID information otherwise
 	rotationCount := k.GetRotationCount(ctx, chain, keyRole) + 1
-	k.setKeyID(ctx, chain, rotationCount, keyRole, keyID)
-	k.setRotationCountOfKeyID(ctx, keyID, rotationCount)
+	_, ok := k.getKeyID(ctx, chain.Name, rotationCount, keyRole)
```

Comment:
is it posible?

Reply:
Yes. If we don't check it here, one will be able to change the next assigned key by calling `AssignNextKey` again.